### PR TITLE
Comment on generics

### PR DESCRIPTION
**Pull request by agnivade**
- Added a section on how to use generics.
- While here, we also remove the comment on the model package
since 6.0 is already released and changes have been made to make
everything consistent.

### CODE REVIEW

**Review by wiggin77 — APPROVED** [review]
Looks great. 👍

File: site/content/contribute/server/style-guide.md [hunk 1]
```diff
@@ -12,17 +12,9 @@ However, at present, some of the guidelines from these sources come into conflic
...
-
 ### Application of guidelines
 
-In addition to the specific exceptions made for backward compatibility in the `model` package, all new commits should also follow the rules as outlined in this and the linked documents, for both new and modified code.
-
-This does not, however, mean that a developer is *required* to fix any surrounding code that contravenes the rules in the style guide. It's encouraged to keep fixing things as you go, but it is not compulsory to do so. Reviewers should refrain from asking for stylistic changes in surrounding code if the submitter has not included them in their pull request.
+The following guidelines should be applied to both new and existing code. This does not, however, mean that a developer is *required* to fix any surrounding code that contravenes the rules in the style guide. It's encouraged to keep fixing things as you go, but it is not compulsory to do so. Reviewers should refrain from asking for stylistic changes in surrounding code if the submitter has not included them in their pull request.
```
Comment (cwarnermm) [inline site/content/contribute/server/style-guide.md:17]:
```suggestion
The following guidelines should be applied to both new and existing code. However, this does not mean that a developer is *required* to fix any surrounding code that contravenes the rules in the style guide. It's encouraged to keep fixing things as you go, but it's not compulsory to do so. Reviewers should refrain from asking for stylistic changes in surrounding code if the submitter has not included them in their pull request.
```

File: site/content/contribute/server/style-guide.md [hunk 2]
```diff
@@ -273,6 +259,12 @@ Any PR that can potentially have a performance impact on the `mattermost-server`
 - Regular expressions, especially when creating `regexp.MustCompile` dynamically every time.
 - Use of the `reflect` package.
 
+### Generics
+
+Generics is a new feature coming in Go 1.18. This is a significant language feature which is yet to be used widely and the patterns around it are unknown. Following the Go team's [recommendation](https://groups.google.com/g/golang-dev/c/iuB22_G9Kbo/m/7B1jd1I3BQAJ), it is advised not to use these features in the main server unless we have more experience from the broader community in using them and some clear patterns emerge.
```
Comment (cwarnermm) [inline site/content/contribute/server/style-guide.md:264]:
```suggestion
Generics is a new feature coming in Go 1.18. This is a significant language feature which is yet to be used widely and the patterns around it are unknown. Following the Go team's [recommendation](https://groups.google.com/g/golang-dev/c/iuB22_G9Kbo/m/7B1jd1I3BQAJ), it's advised not to use these features in the main server unless we have more experience from the broader community in using them and some clear patterns emerge.
```

**Review by cwarnermm — APPROVED** [review]
Good updates! Non-blocking feedback provided inline. Approving for expediency.

**Comment by agnivade** [comment]
Thanks @cwarnermm !

**Pull request merged**